### ISSUE TITLE
File name not showing properly

### ISSUE THREAD

**Issue by olendorf** [psu-libraries/scholarsphere-3#969]
For the work https://scholarsphere.psu.edu/concern/generic_works/9kd17cs76p one or more files don't seem to display properly. 

Also looks like file characterization failed. 

Looks like /concern/parent/9kd17cs76p/file_sets/25425k992r in the file list

and is blank on the fileset show page

https://scholarsphere.psu.edu/concern/parent/9kd17cs76p/file_sets/25425k992r

**Comment by awead** [comment]
ImportUrlJob failed with Resque::DirtyExit: Child process received unhandled signal. They must have been adding files via Browse Everything. They will need to delete the file set and try adding it again. There's no way, I think, to retry this because the url from BE will have expired.

The fix for this should be a change to the ImportUrlJob so that failures do not cause the file set to be added to the work, and instead there's a user notification informing them that the file failed to be added and try again.

**Comment by awead** [comment]
Here's another work with similar problems:

https://scholarsphere.psu.edu/concern/generic_works/4mw22v483z

All these files had some kind of error in Resque, even the ones that were successfully added to the work. I believe this is due to the size of the files. Errors ranged from dirty exits, no memory errors and timeouts from hydra-derivatives.

**Comment by awead** [comment]
Of the file sets that we not successfully added to the work, the commonality was either dirty exit or no memory. The no memory errors (~3) all have stack traces that look like:

``` bash
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/httparty-0.14.0/lib/httparty/request.rb:127:in `join'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/httparty-0.14.0/lib/httparty/request.rb:127:in `block in perform'
/opt/heracles/deploy/.rbenv/versions/2.3.3/lib/ruby/2.3.0/net/http.rb:1446:in `block in transport_request'
/opt/heracles/deploy/.rbenv/versions/2.3.3/lib/ruby/2.3.0/net/http/response.rb:163:in `reading_body'
/opt/heracles/deploy/.rbenv/versions/2.3.3/lib/ruby/2.3.0/net/http.rb:1445:in `transport_request'
/opt/heracles/deploy/.rbenv/versions/2.3.3/lib/ruby/2.3.0/net/http.rb:1407:in `request'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/newrelic_rpm-3.17.1.326/lib/new_relic/agent/instrumentation/net.rb:27:in `block (2 levels) in request_with_newrelic_trace'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/newrelic_rpm-3.17.1.326/lib/new_relic/agent.rb:443:in `disable_all_tracing'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/newrelic_rpm-3.17.1.326/lib/new_relic/agent/instrumentation/net.rb:26:in `block in request_with_newrelic_trace'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/newrelic_rpm-3.17.1.326/lib/new_relic/agent/cross_app_tracing.rb:41:in `tl_trace_http_request'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/newrelic_rpm-3.17.1.326/lib/new_relic/agent/instrumentation/net.rb:23:in `request_with_newrelic_trace'
/opt/heracles/deploy/.rbenv/versions/2.3.3/lib/ruby/2.3.0/net/http.rb:1400:in `block in request'
/opt/heracles/deploy/.rbenv/versions/2.3.3/lib/ruby/2.3.0/net/http.rb:853:in `start'
/opt/heracles/deploy/.rbenv/versions/2.3.3/lib/ruby/2.3.0/net/http.rb:1398:in `request'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/newrelic_rpm-3.17.1.326/lib/new_relic/agent/instrumentation/net.rb:27:in `block (2 levels) in request_with_newrelic_trace'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/newrelic_rpm-3.17.1.326/lib/new_relic/agent.rb:443:in `disable_all_tracing'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/newrelic_rpm-3.17.1.326/lib/new_relic/agent/instrumentation/net.rb:26:in `block in request_with_newrelic_trace'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/newrelic_rpm-3.17.1.326/lib/new_relic/agent/cross_app_tracing.rb:48:in `tl_trace_http_request'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/newrelic_rpm-3.17.1.326/lib/new_relic/agent/instrumentation/net.rb:23:in `request_with_newrelic_trace'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/httparty-0.14.0/lib/httparty/request.rb:118:in `perform'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/httparty-0.14.0/lib/httparty.rb:560:in `perform_request'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/httparty-0.14.0/lib/httparty.rb:486:in `get'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/httparty-0.14.0/lib/httparty.rb:598:in `get'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/bundler/gems/browse-everything-0e335958be2c/lib/browse_everything/retriever.rb:51:in `retrieve'
/opt/heracles/deploy/scholarsphere/releases/20170620101403/app/jobs/import_url_job.rb:66:in `copy_remote_file'
/opt/heracles/deploy/scholarsphere/releases/20170620101403/app/jobs/import_url_job.rb:27:in `block in perform'
/opt/heracles/deploy/scholarsphere/releases/20170620101403/app/jobs/import_url_job.rb:26:in `open'
/opt/heracles/deploy/scholarsphere/releases/20170620101403/app/jobs/import_url_job.rb:26:in `perform'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activejob-4.2.7.1/lib/active_job/execution.rb:32:in `block in perform_now'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:117:in `call'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:555:in `block (2 levels) in compile'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:505:in `call'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:498:in `block (2 levels) in around'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:343:in `block (2 levels) in simple'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/newrelic_rpm-3.17.1.326/lib/new_relic/agent/instrumentation/active_job.rb:46:in `perform'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/newrelic_rpm-3.17.1.326/lib/new_relic/agent/instrumentation/active_job.rb:20:in `block (3 levels) in <top (required)>'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:441:in `instance_exec'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:441:in `block in make_lambda'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:342:in `block in simple'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:497:in `block in around'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:505:in `call'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:498:in `block (2 levels) in around'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:343:in `block (2 levels) in simple'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/i18n-0.8.1/lib/i18n.rb:257:in `with_locale'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activejob-4.2.7.1/lib/active_job/translation.rb:7:in `block (2 levels) in <module:Translation>'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:441:in `instance_exec'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:441:in `block in make_lambda'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:342:in `block in simple'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:497:in `block in around'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:505:in `call'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:498:in `block (2 levels) in around'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:343:in `block (2 levels) in simple'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activejob-4.2.7.1/lib/active_job/logging.rb:23:in `block (4 levels) in <module:Logging>'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/notifications.rb:164:in `block in instrument'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/notifications.rb:164:in `instrument'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activejob-4.2.7.1/lib/active_job/logging.rb:22:in `block (3 levels) in <module:Logging>'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activejob-4.2.7.1/lib/active_job/logging.rb:43:in `block in tag_logger'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/tagged_logging.rb:68:in `block in tagged'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/tagged_logging.rb:26:in `tagged'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/tagged_logging.rb:68:in `tagged'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activejob-4.2.7.1/lib/active_job/logging.rb:43:in `tag_logger'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activejob-4.2.7.1/lib/active_job/logging.rb:19:in `block (2 levels) in <module:Logging>'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:441:in `instance_exec'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:441:in `block in make_lambda'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:342:in `block in simple'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:497:in `block in around'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:505:in `call'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:92:in `__run_callbacks__'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:778:in `_run_perform_callbacks'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:81:in `run_callbacks'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activejob-4.2.7.1/lib/active_job/execution.rb:31:in `perform_now'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activejob-4.2.7.1/lib/active_job/execution.rb:21:in `execute'
/opt/heracles/deploy/scholarsphere/shared/bundle/ruby/2.3.0/gems/activejob-4.2.7.1/lib/active_job/queue_adapters/resque_adapter.rb:46:in `perform'
```

**Comment by carolyncole** [comment]
related to #974. How do we handle issues when background jobs fail

**Comment by awead** [comment]
Fixed in #974
